### ISSUE TITLE
Creating SVG elements without rendering

### ISSUE THREAD

**Issue by pantoninho** [wout/svg.js#480]
Good afternoon,

I am trying to create some SVG elements, but I don't want to draw them right away.
To accomplish this, I've created rectangles using `new SVG.Rect()` instead of `container.rect()`, but I've stumbled upon a bug in Firefox.

This works on Chrome and Safari, but does not work on Firefox:

```
var box = new SVG.Rect()
            .size(width, height)
            .cx(cx)
            .cy(cy)
            .radius(rx || 0, ry || 0)
            .attr(boxAttr(themes.default));

        return box;
```

Firefox's call stack
![image](https://cloud.githubusercontent.com/assets/14220276/14504214/bfea1934-01ab-11e6-8ec2-62b1b1026b76.png)

The same process, but to create a Line instead, works in all three browsers

```
var line = new SVG.Line()
            .plot(x1, x2, y1, y2)
            .stroke({
                width: 1
            })
            .attr(lineAttr(themes.default));

        return line;
```

Thank you to everyone involved in this project,
Pedro Antoninho

**Comment by Fuzzyma** [comment]
And there the bug is again!!
I fixed that about 10 times and it always strikes again...
Getting a BBox of not rendered elements in Firefox throws an error. I worked around this by copying the element in the dom but since our clone function trys to insert the cloned element after the original element it throws an error because the element is not in the dom.

So we have to work around the use of clone in the bbox creation.
Thanks for reporting^^

**Comment by Fuzzyma** [comment]
Fixed in 3eb3209. Note, that in order to get a valid bbox from a node which is not in the dom, we have to move it there temporally. That means, a rect-element which uses its bbox by creation will be cloned and moved to the dom for a moment, even if you dont put it there manually.
Its only a clone and never visisble anywhere. However: constructing the rect in a normal way would deny this overhead of cloning and so on...

We already think about removing the bbox call in the rect constructor. This would ofc remove the whole bbox issue at this place.

**Comment by Lewik** [comment]
I have an svg as template. I want to modify that svg and pass it to another library for render. How to do it without rendering in svg.js?
